src/reconcile/rec_visibility.c

```diff
@@ -129,8 +129,11 @@ __rec_append_orig_value(
         }
     }
 
-    if (tombstone != NULL)
+    if (tombstone != NULL) {
```

Comment:
As per the Coverity, the issue is when the error is raised when allocating memory for Tombstone the memory allocated for `append` is not freed. The fix provided here doesn't fix the error.

You need to free the append in the before `return (ret)` only when the `ret != 0`.